### ISSUE TITLE
Standalone Publisher broken

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#5832]
several problems came up with code currently in
https://github.com/dmwm/CRABServer/tree/master/src/python/Publisher

- [x] it tries to get user DN from username and use it, which is broken (and impossible) in CRIC era
- [x] MasterPublisher and TaskPublish look for configuration in two different files
- [x] need to protect against `tmp/publisher_files/` not be present when it starts
- [x] template PublisherConfig.py should point serviceCert/Key to servicecert/key.pem files, not hoscert
- [x] TaskPublisher relies on X509_USER*env variables for accessing DBS, but those are not set by default env script

**Comment by belforte** [comment]
also:
- [ ] should use put files in /tmp since it will be scratched by a reboot. Publisher mus be able resume where left. Also we need `publisher_files` for possible resubmission of publications and diagnostic

**Comment by belforte** [comment]
code now in my private branch https://github.com/belforte/CRABServer/tree/StandalonePublisher

need to test

**Comment by belforte** [comment]
code now in 3.3.2003.rc1

**Comment by belforte** [comment]
issues I find while testing

- [ ] need in config file: config.General.taskFilesDir = '/data/srv/Publisher_files/'
- [ ] need in config file: config.General.cache_area = 'https://cmsweb-testbed.cern.ch/crabserver/preprod/filemetadata'
- [ ] above one should be avoided by having only one param. in config. pointing to REST host. No need to have REST API name in config. file

**Comment by belforte** [comment]
Now:
 File "/data/srv/TaskManager/3.3.2003.rc2/slc7_amd64_gcc630/cms/crabtaskworker/3.3.2003.rc2/lib/python2.7/site-packages/Publisher/PublisherMaster.py", line 483, in startSlave
subprocess.call(["python", taskPublishScript, "configurationFile=%s"%configurationFile, "taskname=%"%workflow])
NameError: global name 'configurationFile' is not defined

**Comment by belforte** [comment]
- [ ] also found this. Which needs a better message as well
```
Error during metadata retrieving from /crabserver/preprod/filemetadata: url=https://cmsweb-testbed.cern.ch/crabserver/preprod/filemetadata, code=400, reason=Bad Request, headers={'Content-Length': '734', 'X-Error-Http': '400', 'X-Rest-Time': '45024.872 us', 'Server': 'CherryPy/unknown', 'Connection': 'close', 'X-Error-Detail': 'Invalid input parameter', 'CMS-Server-Time': 'D=61891 t=1580414451960885', 'X-Rest-Status': '302', 'Date': 'Thu, 30 Jan 2020 20:00:51 GMT', 'Content-Type': 'text/html;charset=utf-8', 'X-Error-Info': "Incorrect 'taskname' parameter", 'X-Error-Id': '8a52ada218d5a297eb2ae4c422d2022c'}
. publisher.py 200124_224521:belforte_crab_20200124_234517
Exception!
```

**Comment by belforte** [comment]
I have reached the point where MasterPublisher runs, prepares json files in /data/srv/Publisher_files and forks subprocesses to run TaskPublish.py.
Will take stock with a PullRequest, squashin away all comments about the above fixes.

**Comment by belforte** [comment]
working now on TaskPublish. in my branch https://github.com/belforte/CRABServer/tree/fixTaskPublish

next issues:
- [ ] config needs to be made accessible inside mark_good and mark_failed
- [ ] use same naming as in PublisherMaster for CRAB REST API's. Do not have API urls in config.
- [ ] use config to tell prod from preprod, like in PublisherMaster
- [ ] add a --dry mode to allow testing by not writing in DBS
- [ ] look into skipping dataset insertion in DBS, only blockInsert is needed according to Yuyi
- [ ] what is this odd line `    del dataset_config['acquisition_era_name']` ?
- [ ] move logs from `/data/srv/Publisher/taskLogs` to `logs/tasks` subdirectory and separate by user, like in TW

**Comment by belforte** [comment]
as of now it works on some test task.
Let's consolidate with a pull request

**Comment by belforte** [comment]
Runs now, but forks too many processes and logs are a mess. Too hard to debug.
New action list:

- [ ] put a limit on how many slaves can run at same time, do not for one per task blindly
- [ ] logs for slaves writes task name twice and calls itself PublisherMaster but since there is one log per task, it is enough to print the task name once
```
2020-06-27 22:30:13,211:INFO:PublisherMaster:200607_192415:belforte_crab_20200607_212403:Process 200607_192415:belforte_crab_20200607_212403 is starting. PID 18932
```
- [ ] maybe more is needed to get decent logs ?
- [ ] I had to do `pkill python` to stop things, need a better way !

**Comment by belforte** [comment]
found out that last publication time is not correctly used and is never updated. Undiscovered bug in original version.
- [ ] take proper care of using and updationg last publication time

**Comment by belforte** [comment]
Closed via #6091
